src/context/walletContext/walletContext.tsx

```diff
@@ -43,43 +44,25 @@ export const WalletProvider = ({ children }: { children: ReactNode }) => {
     ////// BYOW props
     const { address: wagmiAddress, isConnected: isWagmiConnected } = useAccount()
 
+    ////// User props
+    const { user } = useAuth()
+
     ////// Selected Wallet
     const [selectedWallet, setSelectedWallet] = useState<interfaces.IWallet | undefined>(undefined)  // TODO: this is the var that should be exposed for the app to consume, instead of const { address } = useWallet() anywhere
 
     ////// Wallets
-    const { data: wallets } = useQuery({
-        queryKey: ["wallets", kernelClientAddress, wagmiAddress],
+    const { data: wallets } = useQuery<interfaces.IWallet[]>({
+        queryKey: ["wallets", user?.user.userId],
         queryFn: async () => {
-            /**
-             * TODO: fetch wallets from backend
-             * TODO: 2: Remove fetch & pass user?.account ?
-            */
-            const localPasskeys = PasskeyStorage.list()
-            // const walletsResponse = await fetch('/api/peanut/user/get-wallets')
-            // if (walletsResponse.ok) {
-            //     // receive in backend format
-            //     const { dbWallets }: { dbWallets: interfaces.IDBWallet[] } = await walletsResponse.json()
-            //     // manipulate to frontend format (add connected attribute)
-            //     const wallets: interfaces.IWallet[] = dbWallets.map((dbWallet: interfaces.IDBWallet) => ({
-            //         ...dbWallet,
-            //         connected: false    // this property will be processed into accurate values later in the flow
-            //     }))
-            // }
-            return [
-                // { 
-                //     walletProviderType: interfaces.WalletProviderType.BYOW,
-                //     protocolType: interfaces.WalletProtocolType.EVM,
-                //     connected: false,
-                //     address: '0x7D4c7063E003CeB8B9413f63569e7AB968AF3714'
-                // },
-                ...localPasskeys.map(({ handle, account }) => ({
-                    walletProviderType: interfaces.WalletProviderType.PEANUT,
-                    protocolType: interfaces.WalletProtocolType.EVM,
-                    connected: false,
-                    address: account,
-                    handle
-                }))
-            ]
+            const processedWallets = user?.accounts.filter(
+                account => Object.values(interfaces.WalletProviderType).includes(account.account_type)
+            ).map(account=> ({
+                walletProviderType: account.account_type,
+                protocolType: account.chain,
+                address: account.account_identifier,
+                connected: false
+            }))
+            return processedWallets ? processedWallets : []
```

Comment:
questions:
- Can we keep kernelClientAddress as to be non-breaking to current flow ? (Maybe merge both localStorage account & fetched ones)
- Does `account.account_type` is the same type as walletProviderType ?
- Is `handle` missing ?

Maybe we can store this PR until we get the right data into the account type ?

Reply:
Apologies for late reply !

- Yes kernelClientAddress was just to update that query when someone registered a new wallet and got redirected to home screen.
- Totally you're right, as I said since none of this is yet implemented (fetching login, user fetching), we need to keep that extra source to make the app testable. Sorry if i'm not making sense
- I see it's been added, perfect
- Will need it to be displayed in the profile, but we can grab this from the user probably since for now it's one to one relationship with PWs